### PR TITLE
CLOUDP-197009: [ML3] Add atlas deployments search indexes list

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-197009

#### Local mode
```bash
./bin/atlas deployments search indexes list --collection test --db test --deploymentName local473                                                      3s  1.20.3  16.18.0
ID                         NAME         DATABASE   COLLECTION
65020cf972526b54ea959909   test         test       test
6503052d72526b54ea95990f   testAss      test       test
6502f75b72526b54ea95990b   test2222     test       test
6502f74d72526b54ea95990a   testAndrea   test       test
```

#### Interactive Mode
```bash
./bin/atlas deployments search indexes list                                                                                                                1.20.3  16.18.0
? Database test
? Collection test
? Select a deployment local473
ID                         NAME         DATABASE   COLLECTION
65020cf972526b54ea959909   test         test       test
6503052d72526b54ea95990f   testAss      test       test
6502f75b72526b54ea95990b   test2222     test       test
6502f74d72526b54ea95990a   testAndrea   test       test
```

#### Atlas Mode
```bash
./bin/atlas deployments search indexes list --collection listingsAndReviews --db sample_airbnb --deploymentName testCluste --projectId 64b8d3219c4cdc3f9108e318
ID                         NAME       DATABASE        COLLECTION
65031f1e3bb63801ab9fec1b   myIndex2   sample_airbnb   listingsAndReviews
```

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/deployments/search/indexes/list.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	"github.com/mongodb/mongodb-atlas-cli/internal/mongodbclient"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201007/admin"
```
Comment (tibulca) [inline internal/cli/atlas/deployments/search/indexes/list.go:33]: please merge master + 
```suggestion
	atlasv2 "go.mongodb.org/atlas-sdk/v20230201008/admin"
```

File: internal/cli/atlas/deployments/search/indexes/list_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/deployments/options"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/search"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/mocks"
+	"github.com/mongodb/mongodb-atlas-cli/internal/mongodbclient"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-cli/internal/test"
+	"github.com/stretchr/testify/assert"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201007/admin"
```
Comment (tibulca) [inline internal/cli/atlas/deployments/search/indexes/list_test.go:36]:
```suggestion
	atlasv2 "go.mongodb.org/atlas-sdk/v20230201008/admin"
```

File: internal/cli/atlas/deployments/search/indexes/list.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+			Database:   v.Database,
+		}
+	}
+
+	return out
+}
+
+func ListBuilder() *cobra.Command {
+	opts := &ListOpts{}
+	cmd := &cobra.Command{
```
Comment (tibulca) [inline internal/cli/atlas/deployments/search/indexes/list.go:141]: please extend this [e2e test](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/e2e/atlas/deployments_test.go)  to cover the deployments search indexes list command.
Coordinate with @fmenezes as he is working on https://jira.mongodb.org/browse/CLOUDP-200170

File: internal/cli/atlas/deployments/search/indexes/list.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+		Use:   "list",
+		Short: "List all Atlas Search indexes for a deployment.",
```
Comment (tibulca) [inline internal/cli/atlas/deployments/search/indexes/list.go:143]:
```suggestion
		Short: "List all Atlas Search indexes for a deployment.",
		Aliases: []string{"ls"},
```
for consistency with other list commands

File: internal/cli/atlas/deployments/search/indexes/list.go (hunk 3 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/search/indexes/list.go:141]: after creating index, you can list and verify

File: internal/mongodbclient/search_index.go [hunk 5]
```diff
@@ -25,14 +25,18 @@ import (
...
+	collectionField   = "collection"
+	databaseField     = "database"
 )
 
 var ErrSearchIndexNotFound = errors.New("search Index not found")
 
 type SearchIndex interface {
 	CreateSearchIndex(ctx context.Context, collection string, idx *admin.ClusterSearchIndex) error
 	SearchIndex(ctx context.Context, id string) (*SearchIndexDefinition, error)
+	SearchIndexes(ctx context.Context, coll string) ([]*SearchIndexDefinition, error)
```
Comment (tibulca) [inline internal/mongodbclient/search_index.go:39]: does not have to be addressed in this PR:
we create index using a model from SDK `admin.ClusterSearchIndex` but return models from this package `SearchIndexDefinition`. We should have a consistent contract and use only one of the models (preferably admin.ClusterSearchIndex).

Comment (andreaangiolillo) [reply internal/mongodbclient/search_index.go]: Updated

File: internal/cli/atlas/deployments/search/indexes/list.go (hunk 3 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/deployments/search/indexes/list.go:125]: Updated

**Review by tibulca — APPROVED** [review]

**Review by gssbzn — APPROVED** [review]

**Pull request merged**